Rules/CommonScripts/MapVotesCommon.as

```diff
@@ -0,0 +1,610 @@
+		if (cfg.readIntoArray_string(map_names, "mapcycle"))
+		{
+			const u16 arrayleng = map_names.length();
+			const string currentMap = getMap().getMapName();
+			Random _random(getGameTime());
+
+			// randomize button 1 map name
+			int count = 0;
+			bool done = false;
+			while (!done)
+			{				
+				count++;
+				string temp = map_names[_random.NextRanged(arrayleng)];
+				
+				//test to see if the map filename is inside parentheses and cut it out
+				//incase someone wants to add map votes to a gamemode that loads maps via scripts, eg. Challenge/mapcycle.cfg				 
+				string temptest = temp.substr(temp.length() - 1,temp.length() - 1);
+				if (temptest == ")")
+				{
+					string[] name = temp.split(' (');
+					string mapName = name[name.length() - 1];
+					temp = mapName.substr(0,mapName.length() - 1);
+				}
+
+ 				//lots of cases to be sure we get the most randomization and exit safely
+				if (arrayleng == 2)
+				{
+					button1.filename = map_names[0];
+					button1.shortname = getFilenameWithoutExtension(getFilenameWithoutPath(button1.filename));
+					done = true;
+				}
+				else if (temp != button1.filename && temp != button3.filename && temp != currentMap)
+				{
+					button1.filename = temp;
+					button1.shortname = getFilenameWithoutExtension(getFilenameWithoutPath(temp));
+					done = true;
+				}
+				else if (temp != button1.filename && temp != button3.filename)
+				{
+					button1.filename = temp;
+					button1.shortname = getFilenameWithoutExtension(getFilenameWithoutPath(temp));
+					done = true;
+				}
+				else if (temp != button3.filename)
+				{
+					button1.filename = temp;
+					button1.shortname = getFilenameWithoutExtension(getFilenameWithoutPath(temp));
+					done = true;
+				}
+				else if (count == arrayleng) //safely exit the loop, pure random, probably only 1 map in the pool
+				{
+					button1.filename = map_names[_random.NextRanged(arrayleng)];
+					button1.shortname = getFilenameWithoutExtension(getFilenameWithoutPath(button1.filename));
+					done = true;
+				}
+			}
+
+			//randomize button 3 map name
+			count = 0;
+			done = false; 
+			while (!done)
+			{
+				count++;
+				string temp = map_names[_random.NextRanged(arrayleng)];
+
+				string temptest = temp.substr(temp.length() - 1,temp.length() - 1);
+				if (temptest == ")")
+				{
+					string[] name = temp.split(' (');
+					string mapName = name[name.length() - 1];
+					temp = mapName.substr(0,mapName.length() - 1);
+				}
+
+				if (arrayleng == 2)
+				{
+					button3.filename = map_names[1];
+					button3.shortname = getFilenameWithoutExtension(getFilenameWithoutPath(button3.filename));
+					done = true;
+				}
+				else if (temp != button3.filename && temp != button1.filename && temp != currentMap)
+				{
+					button3.filename = temp;
+					button3.shortname = getFilenameWithoutExtension(getFilenameWithoutPath(temp));
+					done = true;
+				}
+				else if (temp != button3.filename && temp != button1.filename)
+				{
```

Comment:
I recognize that you're trying to not let the last two maps that were voted on be voted on again. Instead at the top of this function after you load the maps. map_names.length() > 4 loops through all the map names and remove the elements that match the last two used.